### ISSUE TITLE
Trigger save even if file is not dirty through API

### ISSUE THREAD

**Issue by xconverge** [Microsoft/vscode#22040]
https://code.visualstudio.com/updates/v1_10#_manually-trigger-save-actions

I read this so I assumed the save() command from the API does the same thing.

https://github.com/Microsoft/vscode/blob/master/src/vs/vscode.d.ts#L124

The comment here makes it seem like it doesn't. Does this call trigger a save even if the document is not dirty?

**Comment by jrieken** [comment]
Yes, the `save` API is slightly different than the save user action. What do you need save without save from the API for? On the UX side we added that to run some on-save commands which in theory you can just run from the ext host w/o save

**Comment by xconverge** [comment]
If you see the referenced issue, we have some vim users who use saving/updating modified date on file to trigger some build scripts elsewhere on the system. We use this save() command to trigger the saving using vim commands.

**Comment by jrieken** [comment]
I might be wrong, @bpasero knows for sure, but I think we don't actually save when the file is not dirty. I believe we just run through the on-will-save handlers and unless a on-will-save handler changes the document we won't touch the file on disk.

**Comment by xconverge** [comment]
If I do File-Save, or ctrl+s on a non dirty file, it DOES save either way/update modified. If I call save() on a clean file, it DOES NOT save. I just tested this

**Comment by adelphes** [comment]
Just FYI, the issue of saving (or, more precisely, not saving) clean files has recently come up in #21681 
Not sure if that issue affects how the API call works, but it seems there are arguments both for and against saving clean (non-dirty) files to disk.

**Comment by bpasero** [comment]
The [`ITextFileService.save()`](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/services/textfile/common/textFileService.ts#L385) method accepts a `ISaveOptions` that can carry a `force` flag. The currently only guy that sets `force` to `true` is our [Save File action](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/files/browser/fileActions.ts#L1426). 

If and only if this flag is set, we actually touch the file on disk if is not dirty to trigger external file watchers. Independent from this flag, since 1.10 we always run save participants on any save action.

I believe we should let extensions decide wether to behave like the save action or not by having a new option in the [save method](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/api/node/mainThreadDocuments.ts#L201) to force saving. 

Other solution would be to always set `force: true`, but not sure about the implications. 

Moving back.

**Comment by xconverge** [comment]
Probably safer to let the extension decide

**Comment by xconverge** [comment]
@bpasero I want to knock this one out real quick, any tips on where you would like the argument to be exposed?

**Comment by bpasero** [comment]
This is coming as new API via https://github.com/microsoft/vscode/issues/178713, currently still proposed.

**Comment by philljones22** [comment]
I want to add this api in my website https://nbiclearanceonline.com.ph/
